src/ts/state/helpers.ts

```diff
@@ -0,0 +1,22 @@
+import { IPluginState, IIgnoreMatch } from "./reducer";
+import { IMatch } from "../interfaces/IMatch";
+import { createDecorationsForMatch } from "../utils/decoration";
+import { DecorationSet } from "prosemirror-view";
+
+export const addMatchesToState = <TMatch extends IMatch>(
+  state: IPluginState<TMatch>,
+  doc: any,
+  matches: TMatch[],
+  ignoreMatch: IIgnoreMatch
+) => {
+  const matchesToApply = matches.filter(ignoreMatch);
```

Comment:
And surely this should be filter(!ignoreMatch) ?

Reply:
As above 😅